internal/registry/obj_selector.go

```diff
@@ -84,11 +92,19 @@ func (o *ObjSelector) NextObject() (*ObjectInfo, error) {
 }
 
 // Resets object selector to start scanning objects from the beginning.
-func (o *ObjSelector) Reset() {
+// After resetting the selector is locked for specified lockTime to prevent
+// concurrency issues.
+func (o *ObjSelector) Reset(lockTime int) bool {
```

Comment:
Correct me. The reason to introduce `lockTime` is that multiple VUs call `Reset()` asynchronously, but we expect it to be called only once. 

Lock timeout should work, but it looks a bit confusing: it is not obvious what time duration to choose for lock. Is one second enough, two, or maybe three? Also `lockTime` is a nanoseconds value, which is not obvious from comment description.

Maybe we can improve this in the future. I have an idea of iteration stages `NextObject(stage int)`.

Consider multiple VUs call `NextObject(1)` and iterating over values. When the cycle is done, test code expects to do one more iteration. Instead of calling `Reset()`, VUs start to call `NextObject(2)` asynchronously, so it does not intersect with first iteration cycle. Then we can remove `Reset` and avoid some magic durations. The only change is that `lastId` would be a map.

Let me know what you think about that.